tests/unit/cmd/scan/test_precommit.py

```diff
@@ -98,3 +101,30 @@ def test_get_merge_branch_from_reflog(monkeypatch):
     monkeypatch.setenv("GIT_REFLOG_ACTION", "merge master")  # Simulate merge
     assert check_is_merge_without_conflict()
     assert get_merge_branch_from_reflog() == "master"
+
+
+@my_vcr.use_cassette("test_emoji_filename")
+def test_precommit_with_emoji_filename(tmp_path, cli_fs_runner):
+    """
+    GIVEN a repository with a staged diff on a file with an emoji in its name
+    WHEN the precommit command is run
+    THEN it executes successfully
+    """
+    # Set up repository
+    repo = Repository.create(tmp_path, initial_branch="main")
```

Comment:
nit: I am not sure why the other tests use `initial_branch="master"`, but `"main"` is the default value for `initial_branch`, so it's not necessary to set it.